tools/Azure.Mcp.Tools.EventGrid/src/Commands/Subscription/SubscriptionListCommand.cs

```diff
@@ -0,0 +1,168 @@
+using Azure.Mcp.Tools.EventGrid.Services;
+
+namespace Azure.Mcp.Tools.EventGrid.Commands.Subscription;
+
+public sealed class SubscriptionListCommand(ILogger<SubscriptionListCommand> logger) : BaseEventGridCommand<SubscriptionListOptions>
+{
+    private const string CommandTitle = "List Event Grid Subscriptions";
+    private readonly ILogger<SubscriptionListCommand> _logger = logger;
+    private readonly Option<string> _topicNameOption = EventGridOptionDefinitions.TopicName;
+    private readonly Option<string> _locationOption = EventGridOptionDefinitions.Location;
+
+    public override string Name => "list";
+
+    public override string Description =>
+        """
+        List event subscriptions for topics with filtering and endpoint configuration. This tool shows all active 
+        subscriptions including webhook endpoints, event filters, and delivery retry policies. Returns subscription 
+        details as JSON array. Requires either --topic (bare topic name) OR --subscription. If only --topic is provided
+        the tool searches all accessible subscriptions for a topic with that name. Optional --resource-group/--location
+        may only be used alongside --subscription or --topic. Examples:\n  --subscription <subId>\n  --subscription <subId> --topic <topicName>\n  --topic <topicName>
+        """;
+
+    public override string Title => CommandTitle;
+
+    public override ToolMetadata Metadata => new() { Destructive = false, ReadOnly = true };
+
+    protected override void RegisterOptions(Command command)
+    {
+        base.RegisterOptions(command);
+        UseResourceGroup(); // Optional resource group filtering
+        command.Options.Add(_topicNameOption);
+        command.Options.Add(_locationOption);
+
+        // Use the helper method from base class to avoid duplicating AZURE_SUBSCRIPTION_ID parsing logic
+        command.Validators.Clear();
+        command.Validators.Add(commandResult =>
+        {
+            var hasSubscription = HasSubscriptionAvailable(commandResult);
+            var hasTopicOption = commandResult.HasOptionResult(_topicNameOption);
+
+            if (!hasSubscription && !hasTopicOption)
+            {
+                commandResult.AddError("Either --subscription or --topic is required.");
+            }
+        });
+    }
+
+    protected override SubscriptionListOptions BindOptions(ParseResult parseResult)
+    {
+        var options = base.BindOptions(parseResult);
+        options.TopicName = parseResult.GetValue(_topicNameOption);
+        options.Location = parseResult.GetValue(_locationOption);
+        return options;
+    }
+
+    public override async Task<CommandResponse> ExecuteAsync(CommandContext context, ParseResult parseResult)
+    {
+        if (!Validate(parseResult.CommandResult, context.Response).IsValid)
+        {
+            return context.Response;
+        }
+
+        var options = BindOptions(parseResult);
+
+        // Validation rules / behaviors:
```

Comment:
These should be in the Validate override, see other commands that override the Validate method for help on that design